### ISSUE TITLE
Loading any chart results in a duplicate execution

### ISSUE THREAD

**Issue by usrme** [usrme/gobarchar#4]
Doesn't matter if it is an [example chart](https://github.com/usrme/gobarchar/blob/946f201554319180a24a485048fbf4e414bcd9b2/examplequery.go) or a random one:

```console
$ go run cmd/gobarchar/main.go
2024/06/25 22:22:14 listening on: 8080
2024/06/25 22:22:20 completed in: 150.333µs
2024/06/25 22:22:20 completed in: 74.625µs
```

I've not touched the project in a long time, so perhaps this is something I knew the reason for, but right now my memory escapes me.

**Comment by usrme** [comment]
Also the case for the [deployed version](https://gobarchar.usrme.xyz/):

```plaintext
2024-06-25T19:57:09.608 app[17816477f21648] ams [info] 2024/06/25 19:57:09 listening on: 8080
2024-06-25T19:58:06.407 app[17816477f21648] ams [info] 2024/06/25 19:58:06 completed in: 70.57µs
2024-06-25T19:58:06.536 app[17816477f21648] ams [info] 2024/06/25 19:58:06 completed in: 74.404µs
```

**Comment by usrme** [comment]
However, the issue doesn't present itself when using `curl`.

**Comment by usrme** [comment]
After adding a bit of code to dump the request:

```diff
diff --git a/chart.go b/chart.go
index da3caed..6ddf909 100644
--- a/chart.go
+++ b/chart.go
@@ -4,6 +4,7 @@ import (
        "fmt"
        "math/rand"
        "net/http"
+       "net/http/httputil"
        "sort"
        "strconv"
        "strings"
@@ -69,6 +70,9 @@ func (a chartData) Less(i, j int) bool { return a[i].Value < a[j].Value }

 func PresentBarChart(examples string) http.HandlerFunc {
        return func(w http.ResponseWriter, r *http.Request) {
+               reqDump, _ := httputil.DumpRequest(r, true)
+               fmt.Printf("REQUEST:\n%s", string(reqDump))
                // Check if there are any query parameters; if not, add random key-value pairs
                if len(r.URL.Query()) == 0 {
                        encodeRandomQuery(r)
```

The problem became obvious:

```console
$ go run cmd/gobarchar/main.go
2024/06/26 14:09:15 listening on: 8080
REQUEST:
GET /?February=14.71&March=22.83&April=28.57&May=22.58&June=23.29&July=20.29&August=9.30&September=9.52 HTTP/1.1
Host: localhost:8080
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7
Accept-Encoding: gzip, deflate, br, zstd
Accept-Language: en-GB,en-US;q=0.9,en;q=0.8
Connection: keep-alive
Dnt: 1
If-None-Match: 357e2e5ea931c096a299ed3e617d32837c98368d
Referer: http://localhost:8080/?February=14.71&March=22.83&April=28.57&May=22.58&June=23.29&July=20.29&August=9.30&September=9.52
Sec-Ch-Ua: "Not/A)Brand";v="8", "Chromium";v="126"
Sec-Ch-Ua-Mobile: ?0
Sec-Ch-Ua-Platform: "macOS"
Sec-Fetch-Dest: document
Sec-Fetch-Mode: navigate
Sec-Fetch-Site: same-origin
Sec-Fetch-User: ?1
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36

2024/06/26 14:09:20 completed in: 180µs
REQUEST:
GET /favicon.ico HTTP/1.1
Host: localhost:8080
Accept: image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8
Accept-Encoding: gzip, deflate, br, zstd
Accept-Language: en-GB,en-US;q=0.9,en;q=0.8
Connection: keep-alive
Dnt: 1
If-None-Match: de2fdae5bbaf89a5f22f91aa4d7dd19d23e8c073
Referer: http://localhost:8080/?February=14.71&March=22.83&April=28.57&May=22.58&June=23.29&July=20.29&August=9.30&September=9.52
Sec-Ch-Ua: "Not/A)Brand";v="8", "Chromium";v="126"
Sec-Ch-Ua-Mobile: ?0
Sec-Ch-Ua-Platform: "macOS"
Sec-Fetch-Dest: image
Sec-Fetch-Mode: no-cors
Sec-Fetch-Site: same-origin
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36

2024/06/26 14:09:20 completed in: 140µs
```

Definitely do not need the additional request for the favicon.

**Comment by usrme** [comment]
Released with version v1.5.1.
